test/suites/edgexfoundry/proxy_test.go

```diff
@@ -123,3 +134,42 @@ func generateCerts(tmpDir string) (string, string) {
 
 	return caCertFile, caKeyFile
 }
+
+// waitCertInstall checks if certificate is fully installed
+// up to a maximum retry number
+func waitCertInstall(t *testing.T, caKey string, maxRetry int) {
+
+	type ResponseBody struct {
+		Data []struct {
+			Key string
+		}
+	}
+
+	res := &ResponseBody{}
+
+	for i := 1; ; i++ {
+		t.Logf("Checking certificate installation. Retry %d/%d", i, maxRetry)
+
+		client := &http.Client{}
+		req, err := http.NewRequest("GET", "http://localhost:8001/certificates", nil)
+		require.NoError(t, err)
+		resp, err := client.Do(req)
```

Comment:
Why not simply use http.Get?

Reply:
Thanks for the feedback. Updated!
I was trying to use the same format in TestTLSCert as TestAddProxyUser:  https://github.com/canonical/edgex-snap-testing/blob/fd6547897170cd31844abb93ec0237ba14ca7582/test/suites/edgexfoundry/proxy_test.go#L63-L71